lib/Loop.lua

```diff
@@ -229,43 +244,33 @@ local function orderSystemsByDependencies(unscheduledSystems: { System })
 	local scheduledSystemsSet = {}
 	local scheduledSystems = {}
 
-	local explore = 1
-	local visited = 2
-
-	while #scheduledSystems < #unscheduledSystems do
-		local index = 1
+	local visited, explored = 1, 2
```

Comment:
I feel like these two variables since last PR on the  scheduler has always been a bit confusing to me. Is this an empirical approach to doing topological sorts?

Reply:
its the same concept as [dfs with colors](https://www.geeksforgeeks.org/detect-cycle-direct-graph-using-colors/#). The primary purpose is to detect a cycle in a graph. Traditionally the variable names are colors, however, for this implementation I thought explored/visited made more sense.